packages/core/src/modules/attendance/attendee-repository.ts

```diff
@@ -143,4 +145,35 @@ export class AttendeeRepositoryImpl implements AttendeeRepository {
       },
     })
   }
+
+  async removeAllSelectionResponsesForSelection(attendanceId: AttendanceId, selectionId: string) {
+    const attendees = await this.db.attendee.findMany({
+      where: {
+        attendanceId,
+      },
+      select: {
+        id: true,
+        selections: true,
+      },
+    })
+
+    const attendeesWithSelections = attendees.filter(
+      (attendee) =>
+        attendee.selections &&
+        typeof attendee.selections === "object" &&
+        Array.isArray(attendee.selections) &&
+        attendee.selections.length > 0
+    ) as { id: AttendeeId; selections: AttendanceSelectionResponse[] }[]
```

Comment:
Same here with the conditions. Also, is the type assertion required? Does typing the function work?

```ts
.filter((thing): thing is { id: AttendeeId; selections: AttendanceSelectionResponse[] } => { ... }
```